xsos

```diff
@@ -2855,6 +2855,7 @@ SYSCTL() {
   __P kernel.random.entropy_avail "[bits] "
   __P kernel.hung_task_panic "[bool] "
   __Pa kernel.hung_task_timeout_secs ""  '{if ($1>0) printf "\"%s\"%s  (secs task must be D-state to trigger)", $1, H0; else printf "\"0\"%s  (khungtaskd disabled)", H0}'
+  __Pa kernel.hung_task_warnings "[0-10] "  '{if ($1==0) printf "\"%s\"%s  (hung has previously occurred for more than 10 times)", $1, H0; else if ($1<10) printf "\%s\%s  (hung has previously occurred more than once)", $1, H0; else if ($1>10) printf "\%s\%s  (Default hung_task_warnings value has changed)", $1, H0; else printf "\%s\%s  (hung has never occurred)", $1, H0}'
```

Comment:
Hey @rhn-jaeshin thanks for the contribution. Sorry for neglecting it for so long. Breaking this line down...

Looking at kernel-doc I see:

> hung_task_warnings:
> 
> The maximum number of warnings to report. During a check interval
> if a hung task is detected, this value is decreased by 1.
> When this value reaches 0, no more warnings will be reported.
> This file shows up if CONFIG_DETECT_HUNG_TASK is enabled.
> 
> -1: report an infinite number of warnings.

While the above doesn't say it, from my checking, it appears to default to `10`.

So it's writable but it also automatically changes based on things happening on the system. What a weird sysctl. My interpretation from that:

* you can set it to `-1` and then it will stay at that static value and all hung-tasks will be reported
* you can set it to `0` and it will stay at that static value and no hung-tasks will be reported
* you can set it to a positive value (or leave at default of `10`), but then it will be automatically decremented by one each time a hung-task occurs ... and they'll be reported each time until it gets to zero, after which point they won't be reported any more

How bizarre. Unless you know the original setting, there's really no way to glean good information from the current value (unless it's `-1`). BLEH. So yeah not sure how to capture that in our guidance, but I'll give it a shot.